### PR TITLE
[IMP] crm : improves convert to opportunity and merge opportunities modals

### PR DESCRIPTION
**Pull request by dbeguin**
Adds various improvements :
- As both modals (crm_merge_opportunities and crm_lead_to_opportunity)
are using quite the same code to merge opportunities,
this duplicated code has been put into a single common class
that can be inherited by both modals.

For Lead to opportunity modal : 
- The user do not have to re-select the crm.lead currently being
  converted to opportunity in field 'opportunity_ids'.

- Field 'opportunity_ids' and 'Customers' moved below radio button
  'Conversion action'.

- Validation error message changed into :
  'Please select at least one lead or opportunity to merge with.'

For merge opportunities modal : 
- Field 'opportunity_ids' moved below 'Customers'

For both modals :
- Field Destination opportunity added : This opportunity will be
  considered as the head opportunity into which the other opportunities
  will be merged with.

- Onchange added on 'opportunities_ids':
  Everytime a crm.lead is added/removed, the destination opportunity is
  updated as the one having the highest confidence level
  (and which has a 'user_id').

- Onchange added on destination opportunity : set the 'user_id' from the
  destination opportunity. Or the opportunity which has the highest
  confidence level (and which has a 'user_id').

- Apply usage of crm.common.merge.opportunity.

Task : https://www.odoo.com/web?#id=1915325&action=327&model=project.task&view_type=form&menu_id=4720

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/crm/wizard/crm_common_merge_opportunities.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        """
+        if self.merge_destination_opportunity_id and self.merge_destination_opportunity_id.user_id:
+            self.user_id = self.merge_destination_opportunity_id.user_id
+        else:
+            highest_opp = self._get_highest_opportunity()
+            if highest_opp:
+                self.user_id = highest_opp.user_id
+
+    @api.onchange('user_id')
+    def _onchange_user(self):
```
Comment (jem-odoo) [inline addons/crm/wizard/crm_common_merge_opportunities.py]: `_onchange_user_id`

File: addons/crm/wizard/crm_lead_to_opportunity_views.xml [hunk 2]
```diff
@@ -99,6 +102,7 @@
             <field name="res_model">crm.lead2opportunity.partner</field>
             <field name="view_mode">form</field>
             <field name="view_id" ref="view_crm_lead2opportunity_partner"/>
+            <field name="context">{'default_opportunity_ids': [(4, active_id, None)]}</field>
```
Comment (jem-odoo) [inline addons/crm/wizard/crm_lead_to_opportunity_views.xml]: `None` is optional

File: addons/crm/wizard/crm_common_merge_opportunities.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            else:
+                user_in_team = False
+            if not user_in_team:
+                values = self.env['crm.lead']._onchange_user_values(self.user_id.id if self.user_id else False)
+                self.team_id = values.get('team_id', False)
+
+    def _get_highest_opportunity(self):
+        """ Retrieves the opportunity that has the highest confidence level amongst the opportunity_ids. """
+        if self.opportunity_ids:
+            def opps_key(opportunity):
```
Comment (jem-odoo) [inline addons/crm/wizard/crm_common_merge_opportunities.py]: maybe we should extract this from `merge_opportinity` and create a method on crm.lead. Will we get problem if one of the 2 implementations change ?

**Comment by tde-banana-odoo** [comment]
Task was canceled . Bisous Davidheu.

**Pull request closed without merging**